src/pages/pricing/index.tsx

```diff
@@ -117,11 +119,30 @@ const Pricing: FunctionComponent<PricingProps> = () => {
           </SelectPlan>
         )}
         {prices.annualPrice && needsEmail && (
-          <EmailForm priceId={prices.annualPrice.price_id} />
+          <EmailForm
+            priceId={prices.annualPrice.price_id}
+            redirectURL={redirectURL}
+          />
         )}
       </div>
     </>
   )
 }
 
+export async function getServerSideProps(context: any) {
+  let redirectURL = null
+  try {
+    const referer = context.req.headers.referer
+    const refererURL = new URL(referer)
+    if (refererURL.origin === process.env.NEXT_PUBLIC_REDIRECT_URI)
+      redirectURL = referer
+  } catch (_e) {
+    // no redirect URL from referer
+  }
```

Comment:
In the context of `getServerSideProps` we can get the referer URL and if it is an egghead-next URL, then it will be treated as a redirect URL to be passed to `stripeCheckoutRedirect`.

Reply:
this is the only part of this I'm not content with. I think we could consider using `useLastResource` to get the redirectURL if it exists.

The problem with this approach is that it is indiscriminate, and in my experience users don't want to be redirected to the last referer and instead the last "point of action" which in our case is always some resource.

I think if we went that route, I'd be fine with the approach described in the [other comment](https://github.com/eggheadio/egghead-next/pull/245#issuecomment-752706204)